src/client/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -123,8 +123,10 @@ public string Claims
         // TODO: ideally, these can come from the particular request instance and not be in RequestBase since it's not valid for all requests.
 
 
-        // TODO: ideally, this can come from the particular request instance and not be in RequestBase since it's not valid for all requests.
-        public bool SendX5C { get; set; }
+        // This should be set on a per-application basis, but can be overridden on a per-request basis should it be needed. 
+        public bool SendX5C { get; set; } 
```

Comment:
Once the request is build / validated, we should know for sure the state of sendX5c. Have a look at AbstractAcquireTokenParameterBuilder.Validate() which can be used for this purpose. We should leak this abstraction into the request variables.